backend/app/readers/all_source_reader.py

```diff
@@ -12,7 +12,7 @@ class AllSourceReader(Reader):
     Read in output data and yield them as Records.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, settings: Settings) -> None:
```

Comment:
Nit: I'd use the `*` more liberally here and elsewhere. In my Python it's practically the default, except in functions where the names don't really matter (like `compare(left, right)` or `max(value1, value2)`).

Reply:
I have added `*` to every class except the `OpenaiAntiRecommender` classes.